llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -568,6 +568,32 @@ AArch64TTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     }
     return Cost;
   }
+  case Intrinsic::vector_extract: {
```

Comment:
I think it would be good to have proper cost model test cases in llvm/test/Analysis/CostModel/AArch64. You can look at existing examples as a guide.